File: packages/mail-composer/ios/Plugin/Classes/Results/CanComposeMailResult.swift
import Foundation
import Capacitor

@objc public class CanComposeMailResult: NSObject, Result {
    let canCompose: Bool

    init(canCompose: Bool) {
        self.canCompose = canCompose
    }

    @objc public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["canCompose"] = canCompose
        return result as AnyObject
    }
}

File: packages/mail-composer/android/src/main/java/io/capawesome/capacitorjs/plugins/mailcomposer/classes/results/CanComposeMailResult.java
package io.capawesome.capacitorjs.plugins.mailcomposer.classes.results;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.mailcomposer.interfaces.Result;

public class CanComposeMailResult implements Result {

    private final boolean canCompose;

    public CanComposeMailResult(boolean canCompose) {
        this.canCompose = canCompose;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("canCompose", canCompose);
        return result;
    }
}
